src/index.js

```diff
@@ -94,19 +95,34 @@ export default class AuthnWidget {
   }
 
   init() {
-    try {
-      if (!this.flowId) {
-        throw new Error(AuthnWidget.FLOW_ID_REQUIRED_MSG);
+    if (window.opener !== null) {
+      // for external authentication in popup, refresh parent window
+      // to get new state and close popup
+      window.opener.location.reload(true);
+      window.close();
+    } else {
+      try {
+        if (!this.flowId) {
+          throw new Error(AuthnWidget.FLOW_ID_REQUIRED_MSG);
+        }
+        this.renderSpinner();
+        this.store
+          .dispatch('GET_FLOW')
+          .catch(() => {this.generalErrorRenderer(AuthnWidget.COMMUNICATION_ERROR_MSG)});
+      } catch (err) {
+        console.error(err);
+        this.generalErrorRenderer(err.message);
       }
-      this.store
-        .dispatch('GET_FLOW')
-        .catch(() => this.generalErrorRenderer(AuthnWidget.COMMUNICATION_ERROR_MSG));
-    } catch (err) {
-      console.error(err);
-      this.generalErrorRenderer(err.message);
     }
   }
 
+  renderSpinner() {
```

Comment:
Can we specify this method is rendering the spinner template and not showing the spinner on the current page?